src/main/java/com/pnu/ordermanagementapp/controller/OrderController.java

```diff
@@ -113,6 +117,29 @@ public String findAllForProduct(@RequestParam(name = "page", required = false, d
         return "orders/show_orders";
     }
 
+    @GetMapping("/export")
+    public String download(
+            @RequestParam("startDate") String startDate,
+            @RequestParam("endDate") String endDate,
+            @RequestParam(name = "clientId", required = false) Long clientId,
+            @RequestParam(name = "productId", required = false) Long productId,
+            Model model, @AuthenticationPrincipal User user) {
```

Comment:
What about State?

Reply:
Hmm
I thought that it wound be enough to provide listing of all orders for some period, sorted by date.
It can be too much filters on one page, so I left it like it is and added state to excel report.